custom_components/webrtc/www/digital-ptz.js

```diff
@@ -0,0 +1,333 @@
+// js version generated from https://github.com/dbuezas/pan-zoom-controller/blob/main/src/digital-ptz.ts
+const ONE_FINGER_ZOOM_SPEED = 1 / 200; // 1 scale every 200px
+const DBL_CLICK_MS = 400;
+const MAX_ZOOM = 10;
+const DEFAULT_OPTIONS = {
+  touch_drag_pan: true,
+  touch_tap_drag_zoom: true,
+  mouse_drag_pan: true,
+  mouse_wheel_zoom: true,
+  mouse_double_click_zoom: true,
+  touch_pinch_zoom: true,
+  persist_key: "",
+  persist: true,
+};
+export class DigitalPTZ {
+  containerEl;
+  videoEl;
+  resizeObserver;
+  transform;
+  options;
+  offHandles = [];
+  constructor(containerEl, videoEl, options) {
+    this.containerEl = containerEl;
+    this.videoEl = videoEl;
+    this.options = Object.assign({}, DEFAULT_OPTIONS, options);
+    this.transform = new Transform({
+      persist_key: this.options.persist_key,
+      persist: this.options.persist,
+    });
+    const o = this.options;
+    const gestureParam = {
+      containerEl: this.containerEl,
+      transform: this.transform,
+      render: this.render,
+    };
+    const h = this.offHandles;
+    if (o.mouse_drag_pan) h.push(startMouseDragPan(gestureParam));
+    if (o.mouse_wheel_zoom) h.push(startMouseWheel(gestureParam));
+    if (o.mouse_double_click_zoom) h.push(startDoubleClickZoom(gestureParam));
+    if (o.touch_tap_drag_zoom) h.push(startTouchTapDragZoom(gestureParam));
+    if (o.touch_drag_pan) h.push(startTouchDragPan(gestureParam));
+    if (o.touch_pinch_zoom) h.push(startTouchPinchZoom(gestureParam));
+    this.videoEl.addEventListener("loadedmetadata", this.recomputeRects);
+    this.resizeObserver = new ResizeObserver(this.recomputeRects);
```

Comment:
The problem must be the`resizeObserver = () => {}` notation. Changing it to `this.resizeObserver 
 = resizeObserver.bind(this)`

Reply:
Went for defining them inside constructor instead

Reply:
You can create function inside observer constructor and call your function from it. This will fix first problem.

But there are some similar problems in another places.

Reply:
Exactly